### PR TITLE
v1.8: remap gops to fixed port to avoid port collision with proxy

### LINKED ISSUE #13400: Agent can't start up: msg="Error while creating daemon" error="listen tcp :40457: bind: address already in use"
The agent is trying over and over before it fails:
```
test-output.log:2020-10-04T21:29:56.420565028Z level=warning msg="Attempt to bind DNS Proxy failed, retrying in 4s" error="listen tcp :40457: bind: address already in use" subsys=fqdn/dnsproxy
test-output.log:2020-10-04T21:30:00.420875556Z level=warning msg="Attempt to bind DNS Proxy failed, retrying in 4s" error="listen tcp :40457: bind: address already in use" subsys=fqdn/dnsproxy
test-output.log:2020-10-04T21:30:04.42116714Z level=warning msg="Attempt to bind DNS Proxy failed, retrying in 4s" error="listen tcp :40457: bind: address already in use" subsys=fqdn/dnsproxy
test-output.log:2020-10-04T21:30:08.421419711Z level=warning msg="Attempt to bind DNS Proxy failed, retrying in 4s" error="listen tcp :40457: bind: address already in use" subsys=fqdn/dnsproxy
test-output.log:2020-10-04T21:30:12.42518694Z level=warning msg="Attempt to bind DNS Proxy failed, retrying in 4s" error="listen tcp :40457: bind: address already in use" subsys=fqdn/dnsproxy
test-output.log:2020-10-04T21:30:16.42229825Z level=fatal msg="Error while creating daemon" error="listen tcp :40457: bind: address already in use" subsys=daemon
```

Example of failed CI run: https://jenkins.cilium.io/job/Cilium-PR-K8s-GKE/2645/

[bf78dc42_K8sIdentity_Identity_expiration_Expiration_of_CiliumIdentity.zip](https://github.com/cilium/cilium/files/5325737/bf78dc42_K8sIdentity_Identity_expiration_Expiration_of_CiliumIdentity.zip)

### PR DESCRIPTION
**Pull request by tklauser**
Manual backport of #14329 to fix port collision of the gops port with the proxy (#13400).

I manually backported the commits from #14329 as they didn't apply cleaning using cherry-picking. I also dropped the commit that updates the client-side gops binary in the runtime images - same as was done for the v1.9 backport in #14419.

See individual commits for details.

Fixes #13400 

```release-note
Remap gops to fixed port to avoid port collision with proxy.
```

### CODE REVIEW

**Comment by tklauser** [comment]
test-backport-1.8

**Comment by tklauser** [comment]
test-1.11-netnext

**Review by pchaigno — APPROVED** [review]

File: Documentation/operations/system_requirements.rst [hunk 1]
```diff
@@ -309,16 +309,20 @@ ICMP 8/0                 egress          ``worker-sg`` (self) health checks
...
 4240/tcp                 cluster health checks (``cilium-health``)
 4244/tcp                 Hubble server
 4245/tcp                 Hubble Relay
 6942/tcp                 operator Prometheus metrics
 9090/tcp                 cilium-agent Prometheus metrics
 9876/tcp                 cilium-agent health status API
-======================== ==========================================
+9890/tcp                 cilium-agent gops server (listening on 127.0.0.1)
+9891/tcp                 operator gops server (listening on 127.0.0.1)
+9892/tcp                 clustermesh-apiserver gops server (listening on 127.0.0.1)
```
Comment (borkmann) [inline Documentation/operations/system_requirements.rst]: I don't see 9892 anywhere defined. Is this expected?

Comment (tklauser) [reply Documentation/operations/system_requirements.rst]: No, that's not expected but a mistake in my manual backporting. The clustermesh-apiserver isn't part of v1.8, but I cherry-picked the doc changes from #14329 without checking. Will correct, thanks for pointing this out.

**Review by borkmann — APPROVED** [review]

**Comment by tklauser** [comment]
test-1.15-4.9

**Comment by tklauser** [comment]
test-1.16-4.9

**Comment by tklauser** [comment]
test-1.17-4.19

**Comment by tklauser** [comment]
test-1.18-4.9

**Comment by tklauser** [comment]
test-1.17-4.19

**Comment by tklauser** [comment]
test-1.17-4.19

**Comment by tklauser** [comment]
test-backport-1.8

**Comment by tklauser** [comment]
test-1.16-4.9

**Comment by tklauser** [comment]
test-1.15-4.9

**Comment by tklauser** [comment]
test-1.14-4.9

**Comment by aanm** [comment]
test-backport-1.8

**Comment by aanm** [comment]
test-upstream-k8s

**Pull request merged**